tests/classification/test_average_precision.py

```diff
@@ -118,3 +119,24 @@ def test_average_precision_differentiability(self, preds, sk_metric, target, num
 )
 def test_average_precision(scores, target, expected_score):
     assert average_precision(scores, target) == expected_score
+
+
+@pytest.mark.parametrize(
```

Comment:
I think it should be possible to just add an additional test case here:
https://github.com/PyTorchLightning/metrics/blob/80f654718bccf9d27094d6f8129d8a11b2041d9b/tests/classification/test_average_precision.py#L70
using this input case:
https://github.com/PyTorchLightning/metrics/blob/591bb2d7f92985efb321b54f96b2628c79e56fe5/tests/classification/inputs.py#L31-L33
(same as the one you already have)

Reply:
The initial idea was to test the same input  `(n_samples, n_classes)` such as for [sklearn](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.average_precision_score.html).

`_input_binary_logits` has a bit differ shape `(NUM_BATCHES, BATCH_SIZE)`

Reply:
sorry, you are right, I did not account that the new test only checks a single batch. Then this should be the correct input:
https://github.com/PyTorchLightning/metrics/blob/591bb2d7f92985efb321b54f96b2628c79e56fe5/tests/classification/inputs.py#L50-L53